theano/tensor/extra_ops.py

```diff
@@ -608,7 +608,16 @@ def repeat(x, repeats, axis=None):
 
     .. versionadded:: 0.6
     """
-    return RepeatOp(axis=axis)(x, repeats)
+    ndim  = x.ndim
+    shape = [x.shape[i] for i in xrange(ndim)]
+    if axis is None:
+        rep_tile = [1 for i in xrange(ndim)] + [repeats] 
+        z = tensor.flatten(tensor.tile(x.reshape(shape+[1]), rep_tile))
+    else:
+        rep_tile = [1 for i in xrange(ndim)]
+        rep_tile[axis] = repeats
+        z = tensor.tile(x, rep_tile)
```

Comment:
That does not seem correct. In NumPy:
```python
>>> x = np.array([[1, 2], [3, 4]])
>>> np.repeat(x, 2, axis=0)
array([[1, 2],
       [1, 2],
       [3, 4],
       [3, 4]])
>>> np.tile(x, [2, 1])
array([[1, 2],
       [3, 4],
       [1, 2],
       [3, 4]])
```
So the shape is right, but the value is not correct.
You probably need to add a new dimension, like in the first case, and then reshape (or partially flatten) the result.